ngraph_bridge/ngraph_builder.cc

```diff
@@ -3186,6 +3186,38 @@ static Status TranslateRankOp(const Node* op, const std::vector<const Tensor*>&,
   return Status::OK();
 }
 
+static Status TranslateRandomUniformOp(
+    const Node* op, const std::vector<const Tensor*>& static_input_map,
+    Builder::OpMap& ng_op_map) {
+  shared_ptr<ng::Node> ng_input;
+  TF_RETURN_IF_ERROR(GetInputNodes(ng_op_map, op, &ng_input));
+
+  auto const_min = ConstructNgNode<ng::op::Constant>(
+      op->name(), ng::element::f32, ng::Shape{}, std::vector<float>{0});
+
+  auto const_max = ConstructNgNode<ng::op::Constant>(
+      op->name(), ng::element::f32, ng::Shape{}, std::vector<float>{1});
+
+  std::vector<int64> shape;
+  TF_RETURN_IF_ERROR(GetStaticInputVector(op, 0, static_input_map, &shape));
+
+  auto const_shape = ConstructNgNode<ng::op::Constant>(
+      op->name(), ng::element::i64, ng::Shape{shape.size()}, shape);
+
+  auto const_use_fixed_seed = ConstructNgNode<ng::op::Constant>(
+      op->name(), ng::element::boolean, ng::Shape{}, std::vector<bool>{true});
+
+  tensorflow::int64 seed{};
+  TF_RETURN_IF_ERROR(GetNodeAttr(op->attrs(), "seed", &seed));
+
+  auto random_uniform = ConstructNgNode<ng::op::RandomUniform>(
+      op->name(), const_min, const_max, const_shape, const_use_fixed_seed,
+      seed);
```

Comment:
@shresthamalik
I wonder if we should mark input0 as static and create a new `const_shape`. Maybe we can just read it as a node and pass it to `RandomUniform`'s constructor.

Of course, if the input node is not const, it will produce dynamic shape output. (but with static input marking it will be a graph split)
https://github.com/NervanaSystems/ngraph/blob/ab440246afb5278347f1cbc76f87fb7ec6968570/src/ngraph/op/experimental/random_uniform.cpp#L84

Reply:
@maxbubula and @shresthamalik : has this comment been resolved?

Reply:
yes. marking as resolved